### ISSUE TITLE
weird envirnomnet variables behavior

### ISSUE THREAD

**Issue by body20002** [pypa/pipenv#5532]
### Is your feature request related to a problem? Please describe.

When I try to run a custom script command defined in [scripts] in the Pipfile, I can't make a temporary environment variable like `SOME_VAR='hi' echo $SOME_VAR` and run the command instead I get an error `Error: the command SOME_VAR=hi (from test) could not be found within PATH.`

but when doing
```toml
[scripts]
test = 'echo $SOME_VAR'
```
```bash
export SOME_VAR='hi'
pipenv run test
```
It works normally and prints hi like you'd expect.
I'm not sure if this is intended or not but it's really annoying and the docs says nothing about this.

### Describe the solution you'd like

Evaluate the environment variables when a command is being invoked if it's possible.
it doesn't have to be like the above example maybe something like 
```toml
[scripts]
test = {call='echo $SOME_VAR', env={SOME_VAR='$(pwd)'}}
```
### Describe alternatives you've considered

I've tried swapping single quotes with double, removing and adding quotes, also tried `$()` <= this thing 
```toml
[scripts]
test = "$(export SOME_VAR='hi') echo $SOME_VAR"
```
and
```toml
[scripts]
test = "SOME_VAR=$(pwd) echo $SOME_VAR"
```
nothing works :/
### Additional context
this is actually what I'm trying to do
```toml
[scripts]
dev = "PROJECT_DIR='$(pipenv --where)' python $PROJECT_DIR/src/main.py && npm run dev --prefix $PROJECT_DIR/src/frontend/"
```
-------------------------------------------------------------------------------

Please run `$ pipenv --support`, and paste the results here. Don't put backticks (`` ` ``) around it! The output already contains Markdown formatting.
<details><summary>$ pipenv --support</summary>

Pipenv version: `'2022.11.11'`

Pipenv location: `'/usr/lib/python3.10/site-packages/pipenv'`

Python location: `'/usr/bin/python'`

OS Name: `'posix'`

User pip version: `'22.3'`

user Python installations found:

  - `3.11.0`: `/home/body20002/.local/share/pyenv/versions/3.11.0/bin/python3`
  - `3.10.8`: `/home/body20002/code/work/whatspy/.venv/bin/python3`
  - `3.10.8`: `/home/body20002/code/work/whatspy/.venv/bin/python`
  - `3.10.8`: `/home/body20002/code/work/whatspy/.venv/bin/python3`
  - `3.10.8`: `/home/body20002/code/work/whatspy/.venv/bin/python`
  - `3.10.8`: `/usr/bin/python3`
  - `3.10.8`: `/usr/bin/python`
  - `3.9.15`: `/home/body20002/.local/share/pyenv/versions/3.9.15/bin/python3`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.10.8',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '6.0.9-zen1-1-zen',
 'platform_system': 'Linux',
 'platform_version': '#1 ZEN SMP PREEMPT_DYNAMIC Wed, 16 Nov 2022 17:00:58 '
                     '+0000',
 'python_full_version': '3.10.8',
 'python_version': '3.10',
 'sys_platform': 'linux'}
```

System environment variables:

  - `ALACRITTY_LOG`
  - `ALACRITTY_SOCKET`
  - `ALACRITTY_WINDOW_ID`
  - `ANDROID_HOME`
  - `BROWSER`
  - `CARGO_HOME`
  - `COLORTERM`
  - `CUDA_CACHE_PATH`
  - `DBUS_SESSION_BUS_ADDRESS`
  - `DESKTOP_SESSION`
  - `DISPLAY`
  - `GNUPGHOME`
  - `GRADLE_USER_HOME`
  - `GTK2_RC_FILES`
  - `GTK_MODULES`
  - `HISTFILE`
  - `HOME`
  - `ICEAUTHORITY`
  - `IPYTHONDIR`
  - `JUPYTER_CONFIG_DIR`
  - `LANG`
  - `LC_CTYPE`
  - `LESS`
  - `LESSHISTFILE`
  - `LOGNAME`
  - `LSCOLORS`
  - `LS_COLORS`
  - `MAIL`
  - `MANPAGER`
  - `MOTD_SHOWN`
  - `MYPY_CACHE_DIR`
  - `NPM_CONFIG_USERCONFIG`
  - `OLDPWD`
  - `PAGER`
  - `PATH`
  - `PDFVIEWER`
  - `PIPENV_ACTIVE`
  - `PIPENV_VENV_IN_PROJECT`
  - `PIPENV_VERBOSITY`
  - `PIP_DISABLE_PIP_VERSION_CHECK`
  - `PIP_PYTHON_PATH`
  - `PS1`
  - `PSQL_HISTORY`
  - `PWD`
  - `PYENV_ROOT`
  - `PYTHONDONTWRITEBYTECODE`
  - `PYTHONSTARTUP`
  - `SHELL`
  - `SHLVL`
  - `SOURCERY_TOKEN`
  - `SQLITE_HISTORY`
  - `TERM`
  - `TERMINAL`
  - `TERM_PROGRAM`
  - `TERM_PROGRAM_VERSION`
  - `TEXMFVAR`
  - `TMUX`
  - `TMUX_PANE`
  - `USER`
  - `VIRTUAL_ENV`
  - `WINDOWID`
  - `WINEPREFIX`
  - `XAUTHORITY`
  - `XCURSOR_PATH`
  - `XDG_CACHE_HOME`
  - `XDG_CONFIG_HOME`
  - `XDG_CURRENT_DESKTOP`
  - `XDG_DATA_HOME`
  - `XDG_RUNTIME_DIR`
  - `XDG_SEAT`
  - `XDG_SEAT_PATH`
  - `XDG_SESSION_CLASS`
  - `XDG_SESSION_DESKTOP`
  - `XDG_SESSION_ID`
  - `XDG_SESSION_PATH`
  - `XDG_SESSION_TYPE`
  - `XDG_STATE_HOME`
  - `XDG_VTNR`
  - `ZDOTDIR`
  - `ZSH`
  - `_`
  - `_JAVA_OPTIONS`
  - `_Z_DATA`
  - `EDITOR`
  - `PYTHONFINDER_IGNORE_UNSUPPORTED`

Pipenv–specific environment variables:

 - `PIPENV_ACTIVE`: `1`
 - `PIPENV_VENV_IN_PROJECT`: `1`
 - `PIPENV_VERBOSITY`: `-1`

</details>

**Comment by matteius** [comment]
I suspect this is because it is being executed in a subshell and so when that exits the environment variables are no longer set.   Have you tried just setting env vars you need in a `.env` file in the project directory?  This file should be local to each developer and not checked in, and in general environment variables can be sensitive and should not be checked into scripts.

**Comment by matteius** [comment]
@body20002 did you see my last response?

**Comment by body20002** [comment]
Yeah I knew about `.env` file, I just wanted to see if this was intended or not,
since there was not documentation about this behavior, I thought you can easily set and use environment variables like I shown.

I don't think the path of the root project directory is something _sensitive_

The real problem is when doing
```yaml
[scripts]
dev = "python ./src/main.py"
``` 
the path doesn't get resolved to the root directory where the `Pipfile` is located, which is very different from how something like `npm run dev` works

package.json
```json
{
  "scripts": {
    "echo": "echo $PWD"
  },
}
```
Pipfile
```yaml
[scripts]
echo = 'echo $PWD'
```
![image](https://user-images.githubusercontent.com/33910565/214977304-3a515d08-5808-40c0-b9ab-4b3a51de7356.png)

if this can be improved by either clarifying the documentation or by supporting getting the root directory of the project or setting env varibales in the `Pipfile`, I'd be happy

**Comment by oz123** [comment]
This seems like a misunderstanding of how Unix shells work.
Export VAR is the right way.
FOO=bar pipenv run ./script.py is executed in a new subshell.
One can use .env file also.
We can definitely emphasize in the docs this information.

**Comment by muelli** [comment]
> Export VAR is the right way.

I'd even go as far as saying that running `env FOO=bar ./script` is better because it is more compatible.

**Comment by body20002** [comment]
you're right, my bad.

But the issues is still here like I explained In https://github.com/pypa/pipenv/issues/5532#issuecomment-1405837086
I think having the same behavior as npm would make things better.

Also I found something and I'm not sure why it behaves this way

```bash
# .env
export foo=$PWD
```

```yaml
# Pipfile
[scripts]
echo = 'echo $foo'
```
running `pipenv run echo` gives the result `$PWD`
shouldn't this be evaluated instead of just giving the string `$PWD`?

**Comment by oz123** [comment]
`.env` are not shell scripts. Depending on the implementation that is reading them, they will behave differently. In pipenv, we use dotenv which apparently does not expand variables.
And honestly, I think it's a good thing. There should be a limit to the amount of indirection. Also, my opinion, explicit is better than implicit.
If your software requires so many variables for configuration you should consider using a configuration file.
But that's just my opinion.
Also, I didn't check how nom works, but I guess it's going to make a lot of Chaos changing the current behavior of pipenv run.
It's a logical behavior if you know that, pipenv run is invoking a subshell.
This should clearly be documented.

**Comment by oz123** [comment]
Thanks for that Kaleb.

**Comment by kalebmckale** [comment]
Sorry for confusion. I was referring to the format of the `.env` file. The script `echo $foo` is working correctly. The `.env` should instead read
```
foo="${PWD}"
```
to get the expected behavior.
